mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h

```diff
@@ -4465,163 +4506,1149 @@ struct Conv2DRewritePattern : public OpRewritePattern<T> {
+        auto gemm = b.create<miopen::TransformOp>(
+            loc, transformedFilterMemRefType, weiGKblockKCYX,
+            transformedFilterAttrs, /*populateBounds=*/true);
+        return gemm;
+      };
+      auto gemmA = getWeiGemmGGemmMGemmN(firtFilterDimName);
+      return gemmA;
+    };
+
+    auto getGemmB = [&]() -> Value {
+      // dim of oob check
+      llvm::DenseSet<int> inputOobCheckDims;
+      // key to dim
+      std::map<StringRef, int> currentKeyToDim;
+      for (unsigned i = 0; i < inputLayoutAttr.size(); ++i) {
+        if (auto strAttr =
+                inputLayoutAttr.getValue()[i].template cast<StringAttr>()) {
+          currentKeyToDim[strAttr.getValue()] = i;
+        }
+      }
+      llvm::SmallVector<StringAttr> firstOutputDimName;
+      auto getInGN0N1CHipWip = [&]() {
+        llvm::SmallVector<StringAttr> &curOutputDimName = firstOutputDimName;
+        llvm::SmallVector<int64_t, 7> transformedShape;
+        llvm::SmallVector<NamedAttribute, 3> transformedAttrs;
+        // gi
+        curOutputDimName.push_back(b.getStringAttr("gi"));
+        transformedShape.push_back(g);
+        llvm::SmallVector<NamedAttribute, 5> gDimAttr{
+            b.getNamedAttr("upper_layer_dimensions",
+                           b.getArrayAttr({b.getI32IntegerAttr(0)})),
+            b.getNamedAttr("upper_layer_names",
+                           b.getArrayAttr({curOutputDimName[0]})),
+            b.getNamedAttr("transformation", b.getStringAttr("PassThrough")),
+            b.getNamedAttr(
+                "lower_layer_dimensions",
+                b.getArrayAttr({b.getI32IntegerAttr(currentKeyToDim["gi"])})),
+            b.getNamedAttr("lower_layer_names",
+                           b.getArrayAttr({b.getStringAttr("gi")}))};
+        // ni
+        curOutputDimName.push_back(b.getStringAttr("n0"));
+        transformedShape.push_back(gemmKBlocks);
+        curOutputDimName.push_back(b.getStringAttr("n1"));
+        transformedShape.push_back(n / gemmKBlocks);
+        llvm::SmallVector<NamedAttribute, 5> nDimAttr{
+            b.getNamedAttr("upper_layer_dimensions",
+                           b.getArrayAttr({b.getI32IntegerAttr(1),
+                                           b.getI32IntegerAttr(2)})),
+            b.getNamedAttr(
+                "upper_layer_names",
+                b.getArrayAttr({curOutputDimName[1], curOutputDimName[2]})),
+            b.getNamedAttr(
+                "dimension_lengths",
+                b.getArrayAttr({b.getI32IntegerAttr(gemmKBlocks),
+                                b.getI32IntegerAttr(n / gemmKBlocks)})),
+            b.getNamedAttr("transformation", b.getStringAttr("UnMerge2")),
+            b.getNamedAttr("parameters",
+                           b.getArrayAttr({
+                               b.getI32IntegerAttr(gemmKBlocks),
+                               b.getI32IntegerAttr(n / gemmKBlocks),
+                           })),
```

Comment:
from the look how `UnMerge2` is used, there are only two parameters, so wondering why `UnMerge` can not be used?